feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -841,50 +846,56 @@ apc_cld_mon_cb(struct ev_loop *loop, ev_timer *timer, int revents)
 		return;
 	}
 
+	for(i=0; i < mgr.status_len; i++) {
+		if(!strncmp(mgr.status_keys[i] , "sec_since_connect", 17 ))
+			conn_since = atoi(mgr.status[i]);
+		LOGI("conn_since=%d, %s: %s", conn_since, mgr.status[i], mgr.status_keys[i]);
+	}
+	/*if cloud conn is false then wait for 60 secs - disconnect timer*/
 	if (mgr.is_connected == false) {
 		if (!ev_is_active(&apc_cld_disconn_timer) &&
-		    !ev_is_pending(&apc_cld_disconn_timer)) {
+		    !ev_is_pending(&apc_cld_disconn_timer) &&
+		    apc_stopped == 0) {
 			LOGI("%s: START apc_cld_disconn_timer, %s",
 			     __func__, state.mode);
 			ev_timer_set(&apc_cld_disconn_timer,
 				     APC_CLD_DISCONN_TIME, 0);
 			ev_timer_start(EV_DEFAULT, &apc_cld_disconn_timer);
 		}
 	}
-	else {
-		if (ev_is_active(&apc_cld_disconn_timer)) {
+	else {	/*if cloud is connected, see if its been active for 30 secs*/
+		if (ev_is_active(&apc_cld_disconn_timer) && conn_since > 30) {
```

Comment:
Everything in this else clause depends on "conn_since > 30".
Suggest making it "else if (conn_since > 30) {" to remove the duplicated tests below.
Suggest that the 30 should have a meaningful MACRO used.